common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbDate.java

```diff
@@ -28,58 +30,131 @@
 
-    private static final ThreadLocal<DateFormat> isoDateFormat = ThreadLocal.withInitial(() ->
-            new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ"));
+    private static String patternDefault = "%s-%s-%sT%s:%s:%s.%s%s";
 
     public TbDate() {
-        super();
+        this.instant = Instant.now();
+        initZonedLocalDateTime();
     }
 
     public TbDate(String s) {
-        super(parse(s));
+        parseInstant(s);
     }
 
-    public TbDate(long date) {
-        super(date);
+    /**
+     * String s = "09:15:30 PM, Sun 10/09/2022";
+     * String pattern = "hh:mm:ss a, EEE M/d/uuuu";
+     * @param s
+     * @param pattern
+     */
+    public TbDate(String s, String pattern, Locale locale) {
+        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern(pattern, locale);
+        LocalDateTime localDateTime = LocalDateTime.parse(s, dateTimeFormatter);
+        ZonedDateTime zonedDateTime = localDateTime.atZone(ZoneId.systemDefault());
+        instant = zonedDateTime.toInstant();
+        initZonedLocalDateTime();
     }
 
-    public TbDate(int year, int month, int date) {
-        this(year, month, date, 0, 0, 0);
+    public TbDate(long dateMilliSecond) {
+        instant = Instant.ofEpochMilli(dateMilliSecond);
+        initZonedLocalDateTime();
     }
 
-    public TbDate(int year, int month, int date, int hrs, int min) {
-        this(year, month, date, hrs, min, 0);
+    public TbDate(int year, int month, int date, String... tz) {
+        this(year, month, date, 0, 0, 0, 0, tz);
     }
 
-    public TbDate(int year, int month, int date,
-                  int hrs, int min, int second) {
-        super(new GregorianCalendar(year, month, date, hrs, min, second).getTimeInMillis());
+    public TbDate(int year, int month, int date, int hrs, int min, String... tz) {
+        this(year, month, date, hrs, min, 0, 0, tz);
     }
 
-    public String toDateString() {
-        DateFormat formatter = DateFormat.getDateInstance();
-        return formatter.format(this);
+    public TbDate(int year, int month, int date, int hrs, int min, int second, String... tz) {
+        this(year, month, date, hrs, min, second, 0, tz);
     }
 
-    public String toTimeString() {
-        DateFormat formatter = DateFormat.getTimeInstance(DateFormat.LONG);
-        return formatter.format(this);
+    public TbDate(int year, int month, int date, int hrs, int min, int second, int secondMilli, String... tz) {
+        this(createDateTimeFromPattern(year, month, date, hrs, min, second, secondMilli, tz));
+    }
+
+    public Instant getInstant() {
+        return instant;
+    }
+
+    public void setInstant(Instant instant) {
+       this.instant = instant;
+    }
+    public ZonedDateTime getZonedDateTime() {
+        return zonedDateTime;
     }
 
+    private void initZonedLocalDateTime() {
+        setZonedDateTime(zoneIdUTC);
+        setLocalDateTime();
+    }
+    public void setZonedDateTime(ZoneId z) {
```

Comment:
This method is only used once. Remove it and inline the body of the method to the initZonedLocalDateTime